### PR TITLE
probe NSM: use latest NSM and k8s libs

### PR DESCRIPTION
**Pull request by matrohon**
Signed-off-by: Mathieu Rohon <mathieu.rohon@orange.com>

### CODE REVIEW

**Comment by matrohon** [comment]
NSM changed its API verisoning from v1 to v1alpha1, and the client lib changed too. Those libs rely on other libs also used by skydvie, but with another version... So We need to update all the necessary lib in skydive in order to use the NSM client lib for the new API.

This would need a big bump to a lot of libs used by skydive, especially k8s libs.

Let's see how the CI react...

**Comment by lebauce** [comment]
run skydive-go-fmt

**Comment by lebauce** [comment]
make skydive-unit-tests

**Comment by lebauce** [comment]
run skydive-unit-tests

**Comment by matrohon** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by matrohon** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-go-fmt

**Comment by lebauce** [comment]
run skydive-go-fmt

**Comment by lebauce** [comment]
run skydive-unit-tests

**Comment by lebauce** [comment]
Merged in https://github.com/skydive-project/skydive/pull/1905

File: topology/probes/nsm/nsm.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 	cc "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/crossconnect"
 	localconn "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
 	remoteconn "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/remote/connection"
-	v1 "github.com/networkservicemesh/networkservicemesh/k8s/pkg/apis/networkservice/v1"
+	v1 "github.com/networkservicemesh/networkservicemesh/k8s/pkg/apis/networkservice/v1alpha1"
 	"github.com/networkservicemesh/networkservicemesh/k8s/pkg/networkservice/clientset/versioned"
```
Comment (hunchback) [inline topology/probes/nsm/nsm.go]: i don't think mapping v1alpha1 to v1 is a good practice, rather you should leave default mapping to v1aloha1 as you may at some point have to support both v1 and v1alpha1 ...

File: topology/probes/nsm/nsm.go [hunk 2]
```diff
@@ -70,30 +71,45 @@ func NewNsmProbe(g *graph.Graph) (*Probe, error) {
 	return probe, nil
 }
 
+func getK8SConfig() (*rest.Config, error) {
```
Comment (hunchback) [inline topology/probes/nsm/nsm.go]: this duplicates code already found in topology/probes/k8s ... better you generalize and reuse the code found there

File: topology/probes/nsm/nsm.go [hunk 3]
```diff
@@ -70,30 +71,45 @@ func NewNsmProbe(g *graph.Graph) (*Probe, error) {
...
-		logging.GetLogger().Errorf("Unable to get in K8S cluster config, make sure the analyzer is running inside a K8S pod", err)
 		return
 	}
 
 	logging.GetLogger().Debugf("NSM: getting NSM client")
 	// Initialize clientset
 	nsmClientSet, err := versioned.NewForConfig(config)
 	if err != nil {
-		logging.GetLogger().Errorf("Unable to initialize the NSM probe", err)
+		logging.GetLogger().Errorf("Unable to initialize the NSM probe: %v", err)
```
Comment (hunchback) [inline topology/probes/nsm/nsm.go]: %s should be good enough

File: topology/probes/nsm/nsm.go [hunk 4]
```diff
@@ -70,30 +71,45 @@ func NewNsmProbe(g *graph.Graph) (*Probe, error) {
...
 		return
 	}
 
 	factory := externalversions.NewSharedInformerFactory(nsmClientSet, 0)
 	genericInformer, err := factory.ForResource(v1.SchemeGroupVersion.WithResource(nsmResource))
 	if err != nil {
-		logging.GetLogger().Errorf("Unable to create the K8S cache factory", err)
+		logging.GetLogger().Errorf("Unable to create the K8S cache factory: %v", err)
```
Comment (hunchback) [inline topology/probes/nsm/nsm.go]: same as above

**Pull request closed without merging**